Puzzlepart/prosjektportalen365
# Improvements to Projecttimeline (tagging, descriptions)

### Your checklist for this pull request


### Description

Added description field and the ability to tag the timeline element and display it on the CallOut.
![image](https://user-images.githubusercontent.com/28678468/190687968-e1178460-4ff5-4e26-81d4-8684c40e0c5b.png)

### New SiteField
- 'Etikett' or tag (GtTag) - Choice field for use in 'Tidslinjeinnhold' list

### Changes to Lists and ContentTypes:
- Tidslinjekonfigurasjon
  - Added category field to categorize timeline elements (this will be useful when grouping for the timeline is added)
- Tidslinjeinnhold
  - Tag and description are now default columns for the 'Tidslinjeinnhold' list

### New changes to Timeline webpart
- Filterpanel
  - Projects are now collapsed by default
  - Tag is added as a filterable property

![image](https://user-images.githubusercontent.com/28678468/190672013-7cdec9d9-2f66-49fe-b8a7-7a96767cb5fa.png)

### Minor changes or additions
- Rounded corners of Timline callout (see before/after underneath)
- Added type of element that causes trouble in error handling for Timeline
- Added better error handling for Resource Allocation if a person is not assigned

| Before| After |
| ------- | ----- |
| ![image](https://user-images.githubusercontent.com/28678468/190687968-e1178460-4ff5-4e26-81d4-8684c40e0c5b.png) | ![image](https://user-images.githubusercontent.com/28678468/190689407-449f0b6b-cfe0-47f0-817f-3731d1e86240.png) |

### How to test

1. Add some tags and/or descriptions
2. See tags and description show up
3. Witness the amazing power of tags and the descriptive textualness


### Relevant issues (if applicable)

#764
